keras/initializers.py

```diff
@@ -284,7 +284,7 @@ def __call__(self, shape, dtype=None):
             raise ValueError(
                 'Identity matrix initializer can only be used for 2D matrices.')
 
-        return self.gain * np.eye(shape[0], shape[1])
+        return self.gain * np.eye(shape[0], shape[1], dtype=dtype)
```

Comment:
I think we could refactor this to call `K.eye` instead. The advantage of `K.eye` is that for TF it will not store the numpy array returned by `np.eye` in the TF graph (so the graph will be smaller).

Reply:
It says in the `Initializers` docs that `If the desired matrix is not square, it pads with zeros on the additional rows/columns`. That would mean calling getting a square matrix from `K.eye` and then padding it.

Something like
```python
def __call__(self, shape, dtype=None):
    eye = K.eye(min(shape), dtype=dtype)
    pad_length = max(shape) - min(shape)
    if shape[0] == shape[1]:
        return eye
    elif shape[0] < shape[1]:
        return K.concatenate([eye, K.zeros((shape[0], pad_length), dtype=dtype)], axis=1)
    else:
        return K.concatenate([eye, K.zeros((pad_length, shape[1]), dtype=dtype)], axis=0)
```

Reply:
`K.eye` should already follow this same behavior (at least it does in TF). So no padding necessary.

Reply:
Wouldn't this mean modifying `K.eye` to accept tuple arguments ?

Reply:
Implicitly I think it already supports that. But maybe not in all backends.

Reply:
Only the `tf.eye` supports the non-square. Currently, all the three `K.eye`s don't support that, thus I proposed #12534.

Reply:
@abhaikollara, #12534 has been merged. Could you replace it with `K.eye((shape[0], shape[1]))`?

Reply:
@taehoonlee Done